### PR TITLE
Removing docker.userEmulation

### PR DESCRIPTION
**Pull request by asp8200**
#1404.
 
 I wonder if this thing with docker.userEmulation and GATK-spark is tested by the CI-tests 🤔

TO-DO: Update changelog. Remove silly tag `foo`.

### CODE REVIEW

**Comment by maxulysse** [comment]
Looks good to me, we might need to figure out what to do for spark

**Comment by asp8200** [comment]
> Looks good to me, we might need to figure out what to do for spark

Ok, but we take care of spark in another PR?

**Comment by maxulysse** [comment]
> > Looks good to me, we might need to figure out what to do for spark
> 
> Ok, but we take care of spark in another PR?

let's make an issue for it then

**Comment by FriederikeHanssen** [comment]
ahhhhhhhhhhh so much time spent on this. This was the only way we could ever get it to work reliably. there are loads of slack threads on it. I think there was even a discussion on the nextflow repo that this will hurt us :D

**Comment by asp8200** [comment]
I removed it because I was getting this: `WARN: Undocumented setting "docker.userEmulation" is not supported any more - please remove it from your config`. I also removed it in dev-bamtofastq.

**Comment by asp8200** [comment]
> Can we ask in the discourse/seqera/nextflow help channel? If this breaks stuff it would be great if there is something superseding it

Sure. How about you ask, Rike?  You seem to know a thing or two about this.

Is Spark only a thing for GATK?

I just tried running this test:
```
nextflow run main.nf -profile test_cache,use_gatk_spark,docker --outdir foo
```
on my branch `asp8200:remove_docker_userEmulation` and the pipeline didn't crash :-) What the problem with Nextflow, Sarek and Spark - something with the work-folder?

**Comment by FriederikeHanssen** [comment]
ok so there actually isn't an issue? spark is tested on CI

**Comment by FriederikeHanssen** [comment]
this is one of those changes though where I would like to see all tests to run

**Comment by asp8200** [comment]
> this is one of those changes though where I would like to see all tests to run

Yes, it would be nice to see all tests pass. How do we trigger them?

**Comment by maxulysse** [comment]
> > this is one of those changes though where I would like to see all tests to run
> 
> Yes, it would be nice to see all tests pass. How do we trigger them?

make a dummy change in a gatk spark file, and the tests should be triggered

**Comment by matthdsm** [comment]
Instead of setting `userEmulation` to `true`, you should set the `runOptions` to
```
docker.runOptions = '--platform=linux/amd64 -e "HOME=${HOME}" -v /etc/passwd:/etc/passwd:ro -v /etc/shadow:/etc/shadow:ro -v /etc/group:/etc/group:ro -v $HOME:$HOME'
```
to get the same behavior.

**Comment by asp8200** [comment]
> Instead of setting `userEmulation` to `true`, you should set the `runOptions` to
> 
> ```
> docker.runOptions = '--platform=linux/amd64 -e "HOME=${HOME}" -v /etc/passwd:/etc/passwd:ro -v /etc/shadow:/etc/shadow:ro -v /etc/group:/etc/group:ro -v $HOME:$HOME'
> ```
> 
> to get the same behavior.

Thanks for the suggesion, @matthdsm. However, I wonder if it is necessary to that those runOptions for docker. I just run
```
nextflow run main.nf -profile test_cache,use_gatk_spark,docker --outdir results
```
with nextflow 24.01.0-edge and docker 20.10.21.

I'll try to trigger the CI-tests for GATK-spark on this PR.

**Comment by asp8200** [comment]
The `joint_germline` test is also failing with Matthias' addition to `docker.runOptions`. Note that `joint_germline` is not only failing for "NF latest" but also for "NF 23.04.0".

File: nextflow.config [hunk 1]
```diff
@@ -190,10 +190,9 @@ profiles {
         charliecloud.enabled   = false
         conda.enabled          = false
         docker.enabled         = true
-        docker.userEmulation   = { params.use_gatk_spark ? false : true }.call()
         podman.enabled         = false
         shifter.enabled        = false
-        docker.runOptions      = '-u $(id -u):$(id -g)'
+        docker.runOptions      = '--platform=linux/amd64 -e "HOME=${HOME}" -v /etc/passwd:/etc/passwd:ro -v /etc/shadow:/etc/shadow:ro -v /etc/group:/etc/group:ro -v $HOME:$HOME -u $(id -u):$(id -g)'
```
Comment (FriederikeHanssen) [inline nextflow.config:195]:
```suggestion
        docker.runOptions      = '-u $(id -u):$(id -g)'
```

File: tests/test_joint_germline.yml [hunk 2]
```diff
@@ -4,6 +4,7 @@
     - germline
     - joint_germline
     - variant_calling
+    - foo
```
Comment (FriederikeHanssen) [inline tests/test_joint_germline.yml:7]:
```suggestion
```

**Comment by maxulysse** [comment]
Let's update CHANGELOG and merge

**Review by maxulysse — APPROVED** [review]

**Review by matthdsm — APPROVED** [review]

**Comment by asp8200** [comment]
Apologies, guys 🙏  I didn't test @matthdsm's suggest with the `docker.runOptions` properly. (As far as I recall, I just test it in `nextflow.config`, but I guess I should have also set it in `test.config` and `cache.config`.)

I got some help from @mahesh-panchal on this, and he recommended setting `docker.runOptions` to
```
-u $(id -u):$(id -g) -v /etc/passwd:/etc/passwd:ro -v /etc/shadow:/etc/shadow:ro -v /etc/group:/etc/group:ro
```

I'm updating the `docker.runOptions` in [this](https://github.com/nf-core/sarek/pull/1426) PR.

**Pull request merged**